### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR provides updates to the developer documentation.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x]  I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

WIP

### CODE REVIEW

File: docs/skill.md [hunk 1]
```diff
@@ -62,14 +60,26 @@ There can be one or more `Task` classes per skill. The developer subclasses abst
...
+
+The developer might want to add other classes on the context level which are shared equally across the `Handler`, `Behaviour` and `Task` classes. To this end the developer can subclass an abstract `SharedClass`. These shared classes are made available on the context level upon initialization of the AEA.
+
+Say, the developer has a class called `SomeClass`
+``` python
+class SomeClass(SharedClass):
+    ...
+```
+
+Then, this class is available on the on the context level like so:
```
Comment (marcofavorito) [inline docs/skill.md]: I'd say "An instance of this class"

Comment (marcofavorito) [reply docs/skill.md]: also, fix typo `on the on the context level`

File: docs/index.md [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-### Open source company
+### Open source technology
 
-We are creating infrastructure for external to use to build their own solutions. 
+We are creating infrastructure for agent developers to use to build their own solutions. 
 
-AEA users are:
+AEA users are, among others:
 
 * Data scientists.
```
Comment (marcofavorito) [inline docs/index.md]: Some suggestions for other categories:

- Students/Academics
- AI/ML Researchers 
- Crypto passionates
- Indie developers

**Comment by DavidMinarsch** [comment]
Thanks @marcofavorito , addressed!

**Pull request merged**